src/shelly_hap_rgb.cpp

```diff
@@ -330,42 +374,59 @@ Status RGBWLight::SetState(const std::string &state_json) {
              "{state: %B, brightness: %d, hue: %d, saturation: %d}", &state,
              &brightness, &hue, &saturation);
 
```

Comment:
state, brightness, hue and saturation may be used uninitialized.
remember to always initialize your variables before scanning them from json,
then validate that required keys were passed. SetState methods of other components.